solc/CommandLineInterface.cpp

```diff
@@ -990,10 +990,22 @@ void CommandLineInterface::handleAst(string const& _argStr)
 		map<ASTNode const*, eth::GasMeter::GasConsumption> gasCosts;
 		// FIXME: shouldn't this be done for every contract?
 		if (m_compiler->runtimeAssemblyItems(m_compiler->lastContractName()))
+		{
+			//NOTE: keep the local variable `ret` to prevent a Heisenbug that could happen when use move assignment.
+			//See: https://github.com/ethereum/solidity/issues/3718 for details.
+			#if defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 4000)
```

Comment:
I think for clarity I'd just have this copying code used always with a big fat comment explaining the reasoning (libc on macos is broken on certain versions).

The problem with this ifdef is that very likely this branch won't be tested (compiled) on the CI and therefore any missing change applied to there will not be caught.

This isn't a speed/memory critical piece of code since it is the CLI and as a such a copy shouldn't hurt too much.

However we could still figure out the reason this move operator fails, because move operators are used extensively in the code and I wonder if it fails in other places too or the data structure returned here has a particular reason for breaking the broken libc.

Reply:
@bit-shift @chriseth do you have any objections to @axic's suggestion? I can revert the change with more comments. I agree with @axic that if the code is not performance critical just copying is better.

Reply:
@alibabe I think @axic's objection to the current solution is reasonable. We'd probably need to create another build plan that uses an older version of XCode / clang / libc++ to also build and test a binary containing the workaround. I'm also fine with applying this in general (and accepting the additional copy).

@axic Does it make sense to merge this regardless of any other potential issues with the move-assignment operator in our code base? Right now we don't have a development environment that we could use to investigate this further.